layers/gpuav/spirv/log_error_pass.cpp

```
@@ -0,0 +1,120 @@
+            new_inst->Fill({pointer_type.Id(), module_.error_payload_variable_id_, spv::StorageClassPrivate});
+            module_.type_manager_.AddVariable(std::move(new_inst), pointer_type);
+        }
+
+        const uint32_t uint32_0_id = module_.type_manager_.GetConstantZeroUint32().Id();
+        const uint32_t constant_id = module_.TakeNextId();
+        {
+            auto new_inst = std::make_unique<Instruction>(8, spv::OpConstantComposite);
+            new_inst->Fill({struct_type.Id(), constant_id, uint32_0_id, uint32_0_id, uint32_0_id, uint32_0_id, uint32_0_id});
+            const Constant& clear_constant = module_.type_manager_.AddConstant(std::move(new_inst), struct_type);
+            error_payload_variable_clear_ = clear_constant.Id();
+        }
+    }
+
+    BasicBlock& block = function.GetFirstBlock();
+    InstructionIt inst_it = block.GetFirstInjectableInstrution();
+
+    block.CreateInstruction(spv::OpStore, {module_.error_payload_variable_id_, error_payload_variable_clear_}, &inst_it);
+}
+
+void LogErrorPass::CreateFunctionCallLogError(Function& function, BasicBlock& block, InstructionIt* inst_it) {
+    GetLinkFunction(link_function_id_, kOfflineFunction);
+
+    const uint32_t stage_info_id = GetStageInfo(function, block, *inst_it);
+
+    const uint32_t function_result = module_.TakeNextId();
+    const uint32_t void_type = module_.type_manager_.GetTypeVoid().Id();
+    block.CreateInstruction(spv::OpFunctionCall, {void_type, function_result, link_function_id_, stage_info_id}, inst_it);
+}
+
+// This is a hard thing to fully get right, as things like OpTerminateInvocation/OpKill can end the invocation, but not the shader.
+// The main thing we are trying to find here is if this is our last chance to report an error message
+bool LogErrorPass::IsShaderExiting(const Instruction& inst) const {
+    return inst.Opcode() == spv::OpReturn || inst.Opcode() == spv::OpReturnValue;
```

If say all fragments within a fragment shader invocation call discard, will we miss the opportunity to log an error?

---

well.... see this is why I am avoiding this, so what we can (and we have ZERO tests for this anywhere) is check for `OpTerminateInvocation` and warn the users... or we could add it before that spot, again, would need to test... will create an issue to track

---

100% a "not worth the time and will just accept some silent errors" for the short term